compiler/rustc_trait_selection/src/traits/mod.rs

```diff
@@ -283,6 +284,38 @@ fn do_normalize_predicates<'tcx>(
 
     debug!("do_normalize_predicates: normalized predicates = {:?}", predicates);
 
+    // The next solver returns nothing for values that contains type error.
+    // And type error should cause failure later. So we skip the consistency check.
+    if !tcx.next_trait_solver_globally() && !predicates.references_error() {
+        let infcx = tcx
+            .infer_ctxt()
+            .with_next_trait_solver(true)
+            .ignoring_regions()
+            .build(TypingMode::non_body_analysis());
+        for (orig_pred, pred_by_old) in original_predicates.into_iter().zip(&predicates) {
+            let inconsistent_pred_by_next =
+                match crate::solve::deeply_normalize::<_, ScrubbedTraitError<'tcx>>(
+                    infcx.at(&cause, elaborated_env),
+                    orig_pred,
+                ) {
+                    Ok(pred_by_next) => {
+                        if pred_by_next == *pred_by_old {
```

Comment:
need to fully resolve the predicates before comparing them. This is causing the error in std. Also just generally move this further to the end of this function?

I think doing it here isn't ideally, but should be good enough for a crater run